### ISSUE TITLE
Grav problems

### ISSUE THREAD

**Issue by markusritschel** [businesstools/docker-grav#1]
Hi,

I wanted to try your grav dockerfile. Grav is actually running, but I get some issues, though:
1) Access via my-site.dkr doesn't work. Instead I can add a port mapping (e.g. "8888:80") in the docker-compose.yml, then I can access the site via http://localhost:8888
2) If I access the site then via http://localhost:8888, then I get a "Grav problems" page, saying "/var/www/html/user/data exists but is not writeable"

When I list all the content in my docker /var/www/html, then the permissions for all directories are set for www-data:www-data, except my user folder (the external volume) which has written 1000:1000.
I already tried changing the permissions for the user directory to www-data using "docker-compose exec web chown -R www-data:www-data /var/www/html/user/" but this gives me then an error in the browser that the connection is not safe.

Can you help me with this?

Best regards.

**Comment by leyyinad** [comment]
Hallo Markus,

thank you for trying to use our image and thank you for your feedback. I need to apologize, this looks like I really missed a couple of things in the readme:

1. This image is to be used with [nginx-proxy](https://github.com/jwilder/nginx-proxy). Once you have that one running and resolve the given domain to it, you should be able to access it.

2. Make sure you're copying or mounting a complete grav/`user`-subtree to `/var/www/html/user` and give it the correct ownership. For instance the images we use usually look like:

```Dockerfile
FROM businesstools/grav:0.3.0

# COPY and RUN different things like setup scripts, composer

COPY --chown=www-data user /var/www/html/user/

# or if you're running an older version of dockerd:
# COPY user /var/www/html/user/
# RUN chown -R www-data /var/www/html/user
```

The `1000:1000` reminds me of an older version of Docker in combination with boot2docker and VirtualBox where it mounted host paths via nfs and you had to mess around with some options and permissions. Today I'm using *Docker for Mac* which mounts host volumes without any problems. In production we don't mount local paths, only docker volumes, which should work as well.

What configuration are you running?

Best of luck,
Daniel

**Comment by markusritschel** [comment]
Hi Daniel,

thanks for your fast reply. I'm using docker on linux with _dockerd version 18.03.1-ce_.
I tried your suggestion with `COPY --chown=www-data user /var/www/html/user/`. But when i just run that container (`docker run -p 80:80 business-grav`) I get a connection error in my browser. When I run it with `docker run -p 80:80 -v user:/var/www/html/user business-grav` I get the _Grav problems_ page again. Am I missing something?

Cheers,
Markus

**Comment by leyyinad** [comment]
So when you run the last command, is there a `/var/www/html/user/data` folder inside your container, who owns it and what permissions does it have?

**Comment by markusritschel** [comment]
Yes, that folder exists. `docker exec <container-name> ls -l user` gives me:

    total 24
    drwxr-xr-x 2 root root 4096 Jun  6 20:07 accounts
    drwxr-xr-x 3 root root 4096 Jun  6 20:07 config
    drwxr-xr-x 2 root root 4096 Jun  6 20:07 data
    drwxr-xr-x 6 root root 4096 Jun  6 20:07 pages
    drwxr-xr-x 9 root root 4096 Jun  6 20:07 plugins
    drwxr-xr-x 6 root root 4096 Jun  6 20:07 themes

In my local view, my local user is the owner:

    drwxr-xr-x 2 markusritschel markusritschel 4,0K Jun  6 21:41 data/

**Comment by leyyinad** [comment]
So if it's owned by `root` and has 755, the `www-data`-user won't be able to write into it. Please note that the `chown`-command in the `Dockerfile` will only set permissions for the container itself. Once you mount some other volume (eg. a local folder) over `user/`, you will see the local machine's permissions. A quick fix might be `chmod a+w user`. This might be a security risk on a production machine, but for local development I don't see a problem here.

Does that help?

**Comment by markusritschel** [comment]
You suggest applying the `chmod a+w user` on my local machine, right? No, this changes the permissions but I still get that _Grav problems_ page...

**Comment by leyyinad** [comment]
Sorry, my mistake, I meant `chmod a+w user/data/` (wrong path).

**Comment by markusritschel** [comment]
Hmm, now I only get a connection error.
